hack/podmansnoop

```diff
@@ -42,41 +43,112 @@ TRACEPOINT_PROBE(sched, sched_process_exit)
 
     events.perf_submit(args, &data, sizeof(data));
     return 0;
 }
+
+static int __submit_arg(struct pt_regs *ctx, void *ptr, struct data_t *data)
+{
+    bpf_probe_read_user(data->argv, sizeof(data->argv), ptr);
+    events.perf_submit(ctx, data, sizeof(struct data_t));
+    return 1;
+}
+
+static int submit_arg(struct pt_regs *ctx, void *ptr, struct data_t *data)
+{
+    const char *argp = NULL;
+    bpf_probe_read_user(&argp, sizeof(argp), ptr);
+    if (argp) {
+        return __submit_arg(ctx, (void *)(argp), data);
+    }
+    return 0;
+}
+
+int syscall__execve(struct pt_regs *ctx,
+    const char __user *filename,
+    const char __user *const __user *__argv,
+    const char __user *const __user *__envp)
+{
+
+    struct task_struct *task = (typeof(task))bpf_get_current_task();
+
+    struct data_t data = {};
+    data.pid = bpf_get_current_pid_tgid() >> 32;
+    data.isArgv = 1;
+
+    bpf_get_current_comm(&data.comm, sizeof(data.comm));
+    __submit_arg(ctx, (void *)filename, &data);
+
+    // skip first arg, as we submitted filename
+    #pragma unroll
+    for (int i = 1; i < MAXARG; i++) {
+        if (submit_arg(ctx, (void *)&__argv[i], &data) == 0)
+             goto out;
+    }
+
+    // handle truncated argument list
+    char ellipsis[] = "...";
+    __submit_arg(ctx, (void *)ellipsis, &data);
+out:
+
+    return 0;
+}
+
+int do_ret_sys_execve(struct pt_regs *ctx)
+{
+    struct task_struct *task = (typeof(task))bpf_get_current_task();
+
+    struct data_t data = {};
+    data.pid = bpf_get_current_pid_tgid() >> 32;
+    bpf_get_current_comm(&data.comm, sizeof(data.comm));
+    events.perf_submit(ctx, &data, sizeof(data));
+    return 0;
+}
 """
 
 def _print_header():
-    print("%-16s %-7s %-7s %-7s %-7s" %
-              ("PCOMM", "PID", "PPID", "TID", "AGE(ms)"))
+    print("%-16s %-7s %-7s %-7s %s" %
+              ("PCOMM", "PID", "PPID", "AGE(ms)", "ARGV"))
 
 buffer = None
+pidToArgv = defaultdict(list)
 
 def _print_event(cpu, data, size): # callback
     """Print the exit event."""
     global buffer
     e = buffer["events"].event(data)
 
-    task = e.task.decode()
-    if task == "3":
+    comm = e.comm.decode()
+    if comm == "3":
         # For absolutely unknown reasons, 'crun' appears as '3'.
-        task = "crun"
+        comm = "crun"
 
-    if task not in ["podman", "crun", "runc", "conmon", "netavark", "aardvark-dns"]:
+    if e.isArgv:
+        pidToArgv[e.pid].append(e.argv)
+        return
+
+    if comm not in ["podman", "crun", "runc", "conmon", "netavark", "aardvark-dns"]:
+        try:
+            del(pidToArgv[e.pid])
+        except Exception:
+            pass
         return
 
     age = (e.exit_time - e.start_time) / 1e6
-    print("%-16s %-7d %-7d %-7d %-7.2f " %
-              (task, e.pid, e.ppid, e.tid, age), end="")
+    argv = b' '.join(pidToArgv[e.pid]).replace(b'\n', b'\\n')
+    print("%-16s %-7d %-7d %-7.2f %s" %
+              (comm, e.pid, e.ppid, age, argv), end="")
```

Comment:
This seems to print the argv as `b'command with args'`, can you remove the unnecessary `b''`?